### PR TITLE
[SPARK-27477][BUILD] Kafka token provider should have provided dependency on Spark

### PR DESCRIPTION
**Pull request by koertkuipers**
## What changes were proposed in this pull request?

Change spark-token-provider-kafka-0-10 dependency on spark-core to be provided

## How was this patch tested?

Ran existing unit tests

### CODE REVIEW

**Comment by koertkuipers** [comment]
> **[Test build #104630 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/104630/testReport)** for PR 24384 at commit [`5e6cebd`](https://github.com/apache/spark/commit/5e6cebd80959ec9d9afa3fcc018c716e60b5de35).
> 
>     * This patch **fails Spark unit tests**.
> 
>     * This patch merges cleanly.
> 
>     * This patch adds no public classes.

weird. the tests all pass on my local machine:
```
All tests passed.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Spark Project Parent POM 3.0.0-SNAPSHOT:
[INFO] 
[INFO] Spark Project Parent POM ........................... SUCCESS [  5.319 s]
[INFO] Spark Project Tags ................................. SUCCESS [ 17.537 s]
[INFO] Spark Project Sketch ............................... SUCCESS [ 40.361 s]
[INFO] Spark Project Local DB ............................. SUCCESS [  7.356 s]
[INFO] Spark Project Networking ........................... SUCCESS [ 52.229 s]
[INFO] Spark Project Shuffle Streaming Service ............ SUCCESS [  8.658 s]
[INFO] Spark Project Unsafe ............................... SUCCESS [ 11.595 s]
[INFO] Spark Project Launcher ............................. SUCCESS [  8.481 s]
[INFO] Spark Project Core ................................. SUCCESS [31:17 min]
[INFO] Spark Project ML Local Library ..................... SUCCESS [ 11.890 s]
[INFO] Spark Project GraphX ............................... SUCCESS [01:36 min]
[INFO] Spark Project Streaming ............................ SUCCESS [06:06 min]
[INFO] Spark Project Catalyst ............................. SUCCESS [11:27 min]
[INFO] Spark Project SQL .................................. SUCCESS [  01:08 h]
[INFO] Spark Project ML Library ........................... SUCCESS [24:35 min]
[INFO] Spark Project Tools ................................ SUCCESS [  1.570 s]
[INFO] Spark Project Hive ................................. SUCCESS [55:36 min]
[INFO] Spark Project REPL ................................. SUCCESS [01:47 min]
[INFO] Spark Project Assembly ............................. SUCCESS [  3.580 s]
[INFO] Kafka 0.10+ Token Provider for Streaming ........... SUCCESS [  7.264 s]
[INFO] Spark Integration for Kafka 0.10 ................... SUCCESS [01:25 min]
[INFO] Kafka 0.10+ Source for Structured Streaming ........ SUCCESS [08:18 min]
[INFO] Spark Project Examples ............................. SUCCESS [ 16.488 s]
[INFO] Spark Integration for Kafka 0.10 Assembly .......... SUCCESS [  1.641 s]
[INFO] Spark Avro ......................................... SUCCESS [01:05 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:34 h
[INFO] Finished at: 2019-04-17T01:59:46-04:00
[INFO] ------------------------------------------------------------------------
```

**Comment by dongjoon-hyun** [comment]
Retest this please.

File: external/kafka-0-10-token-provider/pom.xml [hunk 1]
```diff
@@ -39,6 +39,7 @@
       <groupId>org.apache.spark</groupId>
       <artifactId>spark-core_${scala.binary.version}</artifactId>
       <version>${project.version}</version>
+      <scope>provided</scope>
```
Comment (dongjoon-hyun) [inline external/kafka-0-10-token-provider/pom.xml]: cc @gaborgsomogyi

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by gaborgsomogyi** [comment]
retest this

**Comment by gaborgsomogyi** [comment]
retest this please

File: external/kafka-0-10-token-provider/pom.xml (hunk 1 above)
Comment (gaborgsomogyi) [reply external/kafka-0-10-token-provider/pom.xml]: @dongjoon-hyun thanks for pinging.

**Review by gaborgsomogyi — APPROVED** [review]
LGTM.

**Comment by gaborgsomogyi** [comment]
@koertkuipers AppVeyor can be re-run by you.

**Comment by vanzin** [comment]
Merging to master.

**Pull request closed without merging**